assets/script/refactor.js

```diff
@@ -0,0 +1,75 @@
+class BookList {
+  constructor() {
+    this.form = document.querySelector('#book-form');
+    this.bookListContainer = document.querySelector('#book-list');
+    this.books = [{
+      title: 'Lorem Ipsum',
+      author: 'Testero Testyy',
+    },
+    {
+      title: 'Second Book',
+      author: 'Testero Testyy',
+    }];
+    this.newListBook = {};
+    this.loadBooksFromLocalStorage();
+    this.displayBooks();
+  }
+
+  loadBooksFromLocalStorage() {
+    const storedBooks = localStorage.getItem('books');
+    if (storedBooks) {
+      this.books = JSON.parse(storedBooks);
+    } else {
+      localStorage.setItem('books', JSON.stringify(this.books));
+    }
+  }
+
+  book(title, author) {
+    this.newListBook.title = title;
+    this.newListBook.author = author;
+    return { title, author };
+  }
+
+  displayBooks() {
+    this.bookListContainer.innerHTML = '';
+    this.books.forEach((book, index) => {
+      const li = document.createElement('div');
+      li.className = 'book';
+      li.innerHTML = `<span class="name">&nbsp;"${book.title}"&nbsp; by &nbsp; ${book.author}</span> `;
+      const removeBtn = document.createElement('button');
+      removeBtn.className = 'remove';
+      removeBtn.textContent = 'Remove';
+      removeBtn.setAttribute('data-index', index);
+      li.appendChild(removeBtn);
+      this.bookListContainer.appendChild(li);
+    });
+    this.form.reset();
+  }
+
+  addBook(event) {
+    event.preventDefault();
+    const title = event.target[0].value;
+    const author = event.target[1].value;
+    const newBook = this.book(title, author);
+    this.books.push(newBook);
```

Comment:
- [ ] [OPTIONAL] I'd suggest you use the array method `concat()` instead of `push()` as the `push()` method modifies an array directly, it can introduce unexpected changes in the data that the array represents. This can make it harder to reason about the program's behavior and can introduce bugs. By favoring array methods like `concat()` that do not modify the original array You can write more predictable and easier-to-understand code. 😉
   - The same applies to the `splice()` method you have at line `60` I'd suggest you use the `slice()` method instead.